CONN-10486 SnowflakeSinkServiceV2IT refactor

sfc-gh-mbobowski opened the pull request:
In order to ensure that SSv2 works I would like to test the same ITs as we have for SSv1. This PR fixes a couple of problems with the existing ITs:
- All tests were run twice for password and oAuth authentication. This is fundamentally wrong because authentication method doesn't change anything in the service logic. I extracted a single oAuth test and left rest of them as password based.
- Configuration validation tests were inappropiate for IT level testing so I decided to delete it. It's just to heavy.
- Some of the tests were strongly connected with double buffering logic that was removed recently
- Schematization tests were moved to a a separate file

Of course it is faaar from being perfect but it is good enough to continue development of SSv2.

sfc-gh-snowflakedb-snyk-sa commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/snowflakedb-sca-scanning-public-repo/pr-checks/97d86149-0846-4eb8-b3c7-18b1b9fada2d?refs=1663138e55f4542219d8260b840d4790647bf390&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/snowflakedb-sca-scanning-public-repo/pr-checks/97d86149-0846-4eb8-b3c7-18b1b9fada2d/license?refs=1663138e55f4542219d8260b840d4790647bf390&source=prChecksComment)

sfc-gh-mbobowski commented on src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:
```diff
@@ -1061,393 +901,31 @@ public void testStreamingIngestionWithExactlyOnceSemanticsOverlappingOffsets() t
   }
 
-  @Test
-  public void testSchematizationWithTableCreationAndJsonInput() throws Exception {
```
Moved to `SnowflakeSinkServiceV2SchematizationIT`

sfc-gh-mbobowski commented on src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:
```diff
@@ -1061,393 +901,31 @@ public void testStreamingIngestionWithExactlyOnceSemanticsOverlappingOffsets() t
-  }
-
-  @Test
-  public void testSchematizationSchemaEvolutionWithNonNullableColumn() throws Exception {
```
Moved to `SnowflakeSinkServiceV2SchematizationIT`

sfc-gh-mbobowski commented on src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:
```diff
@@ -1061,393 +901,31 @@ public void testStreamingIngestionWithExactlyOnceSemanticsOverlappingOffsets() t
-  }
-
-  @Test
-  public void snowflakeSinkTask_put_whenJsonRecordCannotBeSchematized_sendRecordToDLQ() {
```
Moved to `SnowflakeSinkServiceV2SchematizationIT`

sfc-gh-mbobowski commented on src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:
```diff
@@ -1538,92 +1016,4 @@ public void testStreamingIngest_multipleChannel_distinctClients() throws Excepti
   }
-
-  @Test
-  void testSkippingOffsetsInSchemaEvolution() throws Exception {
```
Moved to `SnowflakeSinkServiceV2SchematizationIT`

sfc-gh-mbobowski commented on src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:
```diff
@@ -1061,393 +901,31 @@ public void testStreamingIngestionWithExactlyOnceSemanticsOverlappingOffsets() t
-
-  @ParameterizedTest(name = "useOAuth: {0}")
-  @ValueSource(booleans = {true, false})
-  public void testStreamingIngestionValidClientPropertiesOverride(boolean useOAuth)
```
Moved to SnowflakeSinkServiceV2LegacyIT

sfc-gh-mbobowski commented on src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:
```diff
@@ -1061,393 +901,31 @@ public void testStreamingIngestionWithExactlyOnceSemanticsOverlappingOffsets() t
-   */
-  @ParameterizedTest(name = "useOAuth: {0}")
-  @ValueSource(booleans = {true, false})
-  public void testStreamingIngestion_invalidClientPropertiesOverride(boolean useOAuth)
```
Removing. This test is not even doing what it claims it does
```
// Wait 20 seconds here and no flush should happen since the max client lag is 30 seconds
```
max client lag is 1s not 30s. It passes because of the flawed assertion.

sfc-gh-mbobowski commented on src/test/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2IT.java:
```diff
@@ -1061,393 +901,31 @@ public void testStreamingIngestionWithExactlyOnceSemanticsOverlappingOffsets() t
-
-  @Test
-  public void testStreamingIngestionInvalidClientLag() {
-    conn = getConn(false);
```
Removing test - the code does not throw an exception that test asserts. It passes only becuase of the flawed assertion logic.

sfc-gh-dseweryn reviewed (commented):
Thank you for addressing most of the comments I made offline. I would argue that we should keep the OAuth tests intact to have certainty that SSV1 will not stop working in the future.

sfc-gh-mbobowski commented:
> Thank you for addressing most of the comments I made offline. I would argue that we should keep the OAuth tests intact (not reduced to a smoke test) to have certainty that SSV1 will not stop working in the future.

OAuth is an authentication mechanism. I see a value in having a smoke test to validate if it works. I don't see any value in running twice test that e.g. validates what happens when channel was closed before ingestion as it is an orthogonal feature to the authentication itself.

I am not against running more tests on auth but let's be specific about which of them brings value.
